File: backend/app/schemas/file.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.enums import FileStatusEnum, ValidFileTypesEnum


class FileResponse(BaseModel):
    id: uuid.UUID
    vault_id: uuid.UUID
    user_id: uuid.UUID
    original_name: str
    file_type: ValidFileTypesEnum
    mime_type: str | None
    size_bytes: int | None
    page_count: int | None
    status: FileStatusEnum
    error_message: str | None
    total_chunks: int
    total_tokens: int
    file_metadata: dict
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class FileListResponse(BaseModel):
    files: list[FileResponse]
    total: int
    page: int
    page_size: int
    pages: int
    has_next: bool
    has_prev: bool

File: frontend/src/types/file.ts
export type FileStatus = 'PROCESSING' | 'READY' | 'FAILED' | 'FAILED'
export type FileType = 'PDF' | 'IMAGE' | 'DOCX' | 'TXT' | 'NOTE'

export interface VaultFile {
  id: string
  vault_id: string
  user_id: string
  original_name: string
  file_type: FileType
  mime_type: string | null
  size_bytes: number | null
  page_count: number | null
  status: FileStatus
  error_message: string | null
  total_chunks: number
  total_tokens: number
  file_metadata: Record<string, unknown>
  created_at: string
  updated_at: string
}

export interface FileUploadResponse {
  file_id: string
  status: FileStatus
}

export interface FileStatusResponse {
  file_id: string
  status: FileStatus
  total_chunks: number
  error_message?: string | null
}
